modules/core/src/test/scala/lucuma/odb/api/repo/OdbRepoTest.scala

```diff
@@ -73,6 +74,192 @@ trait OdbRepoTest {
       (shared == initial ++ some) && (unshared == initial -- some)
     }
 
+  def shareWithOneUniqueTest[I, A, J, B](
```

Comment:
None of the tests in this file, including the original `sharingTest` fully test the error cases for the case where an attempt is made to share `Id`s for non-existent entities. I have some ideas for testing this, but wasn't sure if I should spend more time on it if much of this will be thrown out in the future. If you think we might keep the `*Repo` classes but give them different persistence, it might be worth adding those tests.

Reply:
I don't expect they will be used in actual production.  The goal I had in mind was just to make the backend work well enough to support Explore development while figuring out a viable GraphQL API.